tests/suites/test_suite_pk.function

```diff
@@ -162,6 +160,131 @@ size_t mbedtls_rsa_key_len_func(void *ctx)
 }
 #endif /* MBEDTLS_RSA_C */
 
+#if defined(MBEDTLS_PSA_CRYPTO_C) && defined(MBEDTLS_PK_HAVE_ECC_KEYS)
+static mbedtls_ecp_group_id ecc_pick_grp_id(void)
+{
+#if defined(MBEDTLS_ECP_LIGHT)
+    return mbedtls_ecp_grp_id_list()[0];
+#elif defined(PSA_WANT_ECC_SECP_R1_192)
+    return MBEDTLS_ECP_DP_SECP192R1;
+#elif defined(PSA_WANT_ECC_SECP_R1_224)
+    return MBEDTLS_ECP_DP_SECP224R1;
+#elif defined(PSA_WANT_ECC_SECP_R1_256)
+    return MBEDTLS_ECP_DP_SECP256R1;
+#elif defined(PSA_WANT_ECC_SECP_R1_384)
+    return MBEDTLS_ECP_DP_SECP384R1;
+#elif defined(PSA_WANT_ECC_SECP_R1_521)
+    return MBEDTLS_ECP_DP_SECP521R1;
+#elif defined(PSA_WANT_ECC_SECP_K1_192)
+    return MBEDTLS_ECP_DP_SECP192K1;
+#elif defined(PSA_WANT_ECC_SECP_K1_224)
+    return MBEDTLS_ECP_DP_SECP224K1;
+#elif defined(PSA_WANT_ECC_SECP_K1_256)
+    return MBEDTLS_ECP_DP_SECP256K1;
+#elif defined(PSA_WANT_ECC_BRAINPOOL_P_R1_256)
+    return MBEDTLS_ECP_DP_BP256R1;
+#elif defined(PSA_WANT_ECC_BRAINPOOL_P_R1_384)
+    return MBEDTLS_ECP_DP_BP384R1;
+#elif defined(PSA_WANT_ECC_BRAINPOOL_P_R1_512)
+    return MBEDTLS_ECP_DP_BP512R1;
+#elif defined(PSA_WANT_ECC_MONTGOMERY_255)
+    return MBEDTLS_ECP_DP_CURVE25519;
+#elif defined(PSA_WANT_ECC_MONTGOMERY_448)
+    return MBEDTLS_ECP_DP_CURVE448;
+#else
+    return 0;
+#endif
+}
+#endif /* defined(MBEDTLS_PSA_CRYPTO_C) && defined(MBEDTLS_PK_HAVE_ECC_KEYS) */
+
+#if defined(MBEDTLS_PSA_CRYPTO_C)
+static int pk_setup_for_type(mbedtls_pk_type_t pk_type, int want_pair,
+                             mbedtls_pk_context *pk, psa_key_type_t *psa_type)
+{
+    int ok = 0;
```

Comment:
Why don't we let this function return `-1` on failure and `0` on success? This would help using `TEST_EQUAL` or `TEST_ASSERT` in the newly added tests below instead of rewriting `if (!condition)`...

Reply:
Why -1? It returns a boolean: true (i.e. 1) on success, 0 on failure.

Reply:
The idea was to use `TEST_EQUAL(<something>, 0)` instead of `TEST_EQUAL(<something>, 1)` or `TEST_ASSERT()` in the newly added test cases, but that's an aesthetic thing.

Reply:
> It returns a boolean: true (i.e. 1) on success, 0 on failure.

And why should it return a boolean when its name does not suggest a boolean, and most other functions return 0 for success and negative for failure? See [our style guide](https://mbed-tls.readthedocs.io/en/latest/kb/development/mbedtls-coding-standards/#return-type).

Using a different return convention for this function seems like a recipe for confusion and I can't see any good reason for it. Am I missing one?

Reply:
The reason I'm doing this is that it is one of our conventions for auxiliary functions in test code where the test function can call `test_fail`. You aren't expected to call such a function with `TEST_ASSERT(aux());` because `aux()` has already asserted: you're expected to call this function with `if (!aux()) goto exit;`.

Yes, it is unfortunate that we have two conflicting conventions. And we could get rid of the has-asserted convention now that `TEST_ASSERT(aux())` doesn't override the error information if `aux()` fails. So I guess we could stop using this convention for new functions? I'm ok to change that if you think that we need to deprecate it. If so we should write this deprecation down somewhere (although admittedly I'm not sure if we ever got around to writing down this convention, which is from around the days updating the knowledge base was a pain).

Reply:
By the time you joined, cascaded `TEST_ASSERT` was fine. It reports the first failure location which is what you almost always want. I'm ok with changing this function if we take the decision that boolean auxiliary functions are a _deprecated_ and not just _not positively advantageous_ convention.

Reply:
> you're expected to call this function with `if (!aux()) goto exit;`

My point exactly: all library functions are supposed to be called with `if (func() != 0) handle_error` so why use the opposite convention for auxiliary test functions?